### DISCUSSION TITLE
Question regarding visibility of Contact Roles in Tenant Contacts

### DISCUSSION THREAD

**Discussion by kraileth** [netbox-community/netbox#12785] [Q&A]
Hello NetBox community,

I learned about the software in the 3.2.x days when looking for an inventory system to manage a couple of servers and a number of containers for an Open Source project that I participate in. While I quickly realized from the demo instance that it's very much overkill for such a use case, the software comes pre-packaged on my primary platform (FreeBSD) and thanks to the accompanying wiki article I had my own instance up and running in just a few minutes.

Despite it being more than I initially wanted, I liked it so much that I stuck around and eventually proposed using it at work. I was able to convince my superiors and we stopped a migration from a self-built solution to GLPI and are now migrating to NetBox instead. I've created a package set (currently we can use neither docker nor pip for policy reasons), installed it on Ubuntu and it's running great. User groups and permissions are all set up, but I've run into an issue that I missed during my software evaluation earlier this year.

I've created contact roles like `owner`, `tech-lead`, `ops` and `billing`. Let's say for tenant X the owner is Alice, technical lead is Bob, Carol is the primary operations contact, Dave the secondary ops and Erin is responsible for billing. Now on the Tenant page I have the _Contacts_ tab which lists all of them. However there seems to be no way to quickly see who's who. One of our employees may need to talk to tech lead about a RAM upgrade because the server is constantly swapping while another one would like to notify ops that we found warnings coming up in the logs starting with their latest application deployment. Unfortunately the Role is not displayed on this page and also cannot be selected in _Configure Table_ (wrong object, I guess?).

So right now our employees either need to know that it's Bob and Carol or e.g. click multiple contacts to look at the bottom in the Assignments table. For cases when you have a lot of contacts this is not really ideal. Also let's say there's customer Y where Frank is both the owner and takes care of billing and Greg is the sole IT guy (i.e. technical lead). Frank is displayed twice, but the information which roles he has again is one click away.

This certainly is no deal breaker but still not exactly as easy to work with as it could, either. As I'm only getting started with seriously using NetBox I wonder how others solve this small problem or if it's a non-issue for most people. Abusing the Contact Groups (which do get displayed) for this seems possible but not an elegant solution IMO due to using redundant info in the DB. Would it maybe make sense to allow adding a Role column in the Tenant's Contacts tab in a future version?

Looking forward for suggestions or discussion.

**Comment by candlerb** [comment]
I think you have a good point here.

The Role is an attribute of the Assignment, for example Alice could be the Owner for Tenant X, the technical lead for site Y,  and the operations contact for Rack Z - see diagram [here](https://docs.netbox.dev/en/stable/features/contacts/). Therefore, at `/tenancy/tenants/1/contacts/`, there is a column for "assignments" which is just a number, and you have to click on it to get the details.

However, I don't see why there couldn't be a column which expands this into a list, like IP addresses on an interface.  Given that you're looking at `/tenancy/tenants/1/contacts/`, the contacts in the table ought to be filtered to just those which have an assignment to tenant 1, and hence the only roles of interest are those which specifically relate the contact to tenant 1.
